src/raft.rs

```diff
@@ -1503,22 +1503,27 @@ impl<T: Storage> Raft<T> {
     // step_candidate is shared by state Candidate and PreCandidate; the difference is
     // whether they respond to MsgRequestVote or MsgRequestPreVote.
     fn step_candidate(&mut self, m: Message) -> Result<()> {
-        let term = self.term;
         match m.get_msg_type() {
             MessageType::MsgPropose => {
-                info!("{} no leader at term {}; dropping proposal", self.tag, term);
+                info!(
+                    "{} no leader at term {}; dropping proposal",
+                    self.tag, self.term
+                );
                 return Err(Error::ProposalDropped);
             }
             MessageType::MsgAppend => {
-                self.become_follower(term, m.get_from());
+                debug_assert!(self.term == m.get_term());
```

Comment:
Why not use `debug_assert_eq`?

Reply:
thanks @BusyJay updated all, debug_assert_eq is more suitable.